code/migration/src/main/scala/com/namely/chiefofstate/migration/CreateSchemas.scala

```diff
@@ -0,0 +1,132 @@
+import akka.Done
+import com.github.ghik.silencer.silent
+import com.typesafe.config.Config
+import org.slf4j.{Logger, LoggerFactory}
+import slick.basic.DatabaseConfig
+import slick.jdbc.{JdbcProfile, PostgresProfile}
+
+import java.sql.{Connection, Statement}
+import scala.concurrent.{ExecutionContextExecutor, Future}
+
+@silent
+object CreateSchemas {
+  final val log: Logger = LoggerFactory.getLogger(getClass)
+
+  /**
+   * returns the event_journal and event_tag tables ddl
+   *
+   * @return the sql statement
+   */
+  private def journalTableDDL(): Seq[String] = {
+    Seq(
+      s"""
+        CREATE TABLE IF NOT EXISTS event_journal(
+          ordering BIGSERIAL,
+          persistence_id VARCHAR(255) NOT NULL,
+          sequence_number BIGINT NOT NULL,
+          deleted BOOLEAN DEFAULT FALSE NOT NULL,
+
+          writer VARCHAR(255) NOT NULL,
+          write_timestamp BIGINT,
+          adapter_manifest VARCHAR(255),
+
+          event_ser_id INTEGER NOT NULL,
+          event_ser_manifest VARCHAR(255) NOT NULL,
+          event_payload BYTEA NOT NULL,
+
+          meta_ser_id INTEGER,
+          meta_ser_manifest VARCHAR(255),
+          meta_payload BYTEA,
+
+          PRIMARY KEY(persistence_id, sequence_number)
+        );""",
+      // create index
+      s"""CREATE UNIQUE INDEX event_journal_ordering_idx ON event_journal(ordering);""",
+      // create the event_tag
+      s"""
+        CREATE TABLE IF NOT EXISTS event_tag(
+            event_id BIGINT,
+            tag VARCHAR(256),
+            PRIMARY KEY(event_id, tag),
+            CONSTRAINT fk_event_journal
+              FOREIGN KEY(event_id)
+              REFERENCES event_journal(ordering)
+              ON DELETE CASCADE
+        );
+      """
+    )
+  }
+
+  /**
+   * return the snapshot ddl statement
+   * @return the sql statement
+   */
+  private def snapshotTableDDL(): String = {
+    s"""
+     CREATE TABLE IF NOT EXISTS state_snapshot (
+      persistence_id VARCHAR(255) NOT NULL,
+      sequence_number BIGINT NOT NULL,
+      created BIGINT NOT NULL,
+
+      snapshot_ser_id INTEGER NOT NULL,
+      snapshot_ser_manifest VARCHAR(255) NOT NULL,
+      snapshot_payload BYTEA NOT NULL,
+
+      meta_ser_id INTEGER,
+      meta_ser_manifest VARCHAR(255),
+      meta_payload BYTEA,
+
+      PRIMARY KEY(persistence_id, sequence_number)
+     )"""
+  }
+
+  /**
+   * creates the various write-side stores and read-side offset stores
+   *
+   * @param config the application config
+   */
+  def ifNotExists(config: Config)(implicit system: ActorSystem[_]): Future[Done] = {
+
+    implicit val ec: ExecutionContextExecutor = system.executionContext
+
+    val dbconfig: DatabaseConfig[JdbcProfile] = JdbcConfig.getWriteSideConfig(config)
+    val readSideJdbcConfig: DatabaseConfig[PostgresProfile] = JdbcConfig.getReadSideConfig(config)
+
+    // let us get the database connection
+    val conn: Connection = dbconfig.db.createSession().conn
+    try {
```

Comment:
let's rewrite these Try's to better handle failure. I want to be sure it fails and quits if schema creation fails.

Reply:
As one can see I did not handle the exceptions. The goal is to let-it-crash when there is an error because since the StartNodeBehaviour is the `user guardian actor` it will stop the whole actor system in case of exception because that actor is stopped in case of error. Reference [Root Actor](https://doc.akka.io/docs/akka/current/general/supervision.html#user-the-user-guardian-actor)

Reply:
So we can explicitly set the supervisory strategy mode of that actor to die in case of exception as stated https://doc.akka.io/docs/akka/2.5/general/supervision.html#user-the-guardian-actor

Reply:
I am changing it to use the built-in akka slick for better connection management